include/boost/math/common_factor.hpp

```diff
@@ -10,10 +10,14 @@
 #ifndef BOOST_MATH_COMMON_FACTOR_HPP
 #define BOOST_MATH_COMMON_FACTOR_HPP
 
+#ifndef BOOST_MATH_STANDALONE
 #include <boost/math/common_factor_ct.hpp>
 #include <boost/math/common_factor_rt.hpp>
+#include <boost/math/tools/header_deprecated.hpp>
 
-BOOST_HEADER_DEPRECATED("<boost/integer/common_factor.hpp>");
-
+BOOST_MATH_HEADER_DEPRECATED("<boost/integer/common_factor.hpp>");
+#else
+#error Common factor is not available in standalone mode because it requires boost.integer.
```

Comment:
Isn't gcd part of the standard library now?

Reply:
Yes; `std::gcd` was added in C++17.